web/massastation/src/pages/Store/Install.tsx

```diff
@@ -23,15 +34,27 @@ function Install() {
 
   function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
     e.preventDefault();
-    const formObject = parseForm(e);
-    if (!validate(formObject)) return;
+    if (!validate(url)) return;
+    const params = { source: url };
+    mutate({ params });
   }
 
   function handleChange(e: any) {
     setError('');
     setUrl(e.target.value);
   }
 
+  useEffect(() => {
+    if (installError) {
+      setError(Intl.t('store.sidepanel.be-error'));
+    }
+  }, [installError]);
+  useEffect(() => {
```

Comment:
Please, split things... BTW we need a lit for this code organisation and spaces.

```suggestion
  }, [installError]);
  
  useEffect(() => {
```